src/api/app/models/notification.rb

```diff
@@ -3,9 +3,30 @@ class Notification < ApplicationRecord
 
   serialize :event_payload, JSON
 
+  def initialize(params = {})
+    super
+    shorten_payload_if_necessary
+  end
+
   def event
     @event ||= event_type.constantize.new(event_payload)
   end
+
+  private
+
+  def shorten_payload_if_necessary
```

Comment:
I'm fine with the changes made to `event/base.rb` but I would really prefer not to merge these changes to `Notification`. We are duplicating the payload shortening code here and its not necessary, better would be to use the exact same encoding method and the exact same mysql column length for events.payload and notifications.event_payload. Doing it that way means that you only have to shorten the payload once (for the events table) and then you already know that the payload can be copied to the notifications table because they use the same encoding.
 
Let me know if you have an disagreements..